### PR TITLE
Custom omega cool console with logs and shit

### PR DESCRIPTION
**Pull request by mufty**
Needs: https://github.com/MafiaHub/Framework/pull/70

Review C++ shit cuz it sucks

### CODE REVIEW

**Comment by mufty** [comment]
image

**Comment by mufty** [comment]
image

**Comment by mufty** [comment]
image

**Comment by mufty** [comment]
F8 opens the thing

File: code/client/src/core/application.h [hunk 1]
```diff
@@ -13,11 +12,10 @@ namespace MafiaMP::Core {
     class Application: public Framework::Integrations::Client::Instance {
       private:
         Framework::Utils::States::Machine *_stateMachine;
+        UI::Console *_console;
```
Comment (zaklaus) [inline code/client/src/core/application.h:15]: use `std::unique_ptr<UI::Console>`

C++ sucks

File: code/client/src/core/ui/console.h [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+namespace MafiaMP::Core::UI {
+    class Console {
+      private:
+        bool _shouldDisplayWidget;
+        bool _isOpen;
+        bool _autoScroll;
+        std::vector<Game::Streaming::EntityTrackingInfo *> _TEMP_vehicles;
+        Framework::Utils::States::Machine *_machine;
+        void disconnect();
```
Comment (zaklaus) [inline code/client/src/core/ui/console.h:20]: PascalCase methods

**Review by zaklaus — CHANGES_REQUESTED** [review]

File: code/client/src/core/application.h (hunk 1 above)
Comment (mufty) [reply code/client/src/core/application.h]: dealt with

File: code/client/src/core/ui/console.h (hunk 2 above)
Comment (mufty) [reply code/client/src/core/ui/console.h]: dealt with

**Review by zaklaus — APPROVED** [review]

**Pull request merged**